### ISSUE TITLE
Checks Ok then BUILD FAILED in Android

### ISSUE THREAD

**Issue by dyguests** [shyiko/ktlint#264]
I add ktlint to my Android project.
when i run `gradle ktlint`,it checks ok but then return BUILD FAILED.

Log:
	apple$ gradle ktlint

	> Task :ktlint FAILED
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:9:1: Wildcard import (cannot be auto-corrected)
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:8:1: Needless blank line(s)
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:17:5: Missing space after //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:18:5: Missing space after //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:19:5: Missing space after //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:20:12: Missing spacing before "="
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/test/java/com/fanhl/ktlintdemo/ExampleUnitTest.kt:5:1: Wildcard import (cannot be auto-corrected)

	FAILURE: Build failed with an exception.

	* What went wrong:
	Execution failed for task ':ktlint'.
	> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

	* Try:
	Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

	* Get more help at https://help.gradle.org

	BUILD FAILED in 2s
	1 actionable task: 1 executed
	lindeiMac:KtlintDemo apple$ 

I tried way in `#170` but not working.
In `#244`,I used Gradle.I have no idea that how to use `ktlint --apply-to-idea-project --android `.

**Add:**

I just run `ktlint` success in `command line`.but not working fine when run `gradle ktlint` with `Gradle`.

**Comment by shyiko** [comment]
Do you mean you're not getting lint errors when running ktlint on the command line (inside `/Users/apple/AndroidStudioProjects/KtlintDemo/app`) and yet `gradle ktlint` inside the same directory fails? If so - would it be possible to attach MainActivity.kt? (gist would be perfect)

**Comment by dyguests** [comment]
Mac
Android Studio 3.1.2
kotlin_version 1.2.31
com.android.tools.build:gradle:3.1.3
gradle distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
compileSdkVersion 27
targetSdkVersion 27

`MainActivity.kt`

https://github.com/dyguests/KtlintDemo/blob/2d8dca342061afc840a550050c2bccbf5ef579fb/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt

`root/build.gradle`

https://github.com/dyguests/KtlintDemo/blob/94a1972a66b37f77424d13658f3a1ba24a9f9049/build.gradle

------------------------
On Terminal:

run `ktlint` works good.

	$ ktlint
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:9:1: Wildcard import (cannot be auto-corrected)
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:23:1: Needless blank line(s)
	...
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:117:7: Missing space after //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/test/java/com/fanhl/ktlintdemo/ExampleUnitTest.kt:5:1: Wildcard import (cannot be auto-corrected)

run `gradle ktlint` work bad.

	$ gradle ktlint
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:9:1: Wildcard import (cannot be auto-corrected)
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:23:1: Needless blank line(s)
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/ExampleInstrumentedTest.kt:25:1: Unexpected blank line(s) before "}"
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/androidTest/java/com/fanhl/ktlintdemo/
	...
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:117:7: Missing space before //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/main/java/com/fanhl/ktlintdemo/MainActivity.kt:117:7: Missing space after //
	/Users/apple/AndroidStudioProjects/KtlintDemo/app/src/test/java/com/fanhl/ktlintdemo/ExampleUnitTest.kt:5:1: Wildcard import (cannot be auto-corrected)


	FAILURE: Build failed with an exception.

	* What went wrong:
	Execution failed for task ':ktlint'.
	> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

	* Try:
	Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

	* Get more help at https://help.gradle.org

	BUILD FAILED in 8s
	1 actionable task: 1 executed

:)

**Comment by Rolnire** [comment]
I have the same result.. I got bunch of checkstyle warnings which is good but at the end **BUILD FAILED** why ?

Warnings for instance:
```
/FlavorModule.kt:1:1: File must end with a newline (\n)
/FlavorModule.kt:1:1: File must end with a newline (\n)


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ktlint'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed
```

Also i dont understand whats wrong with this. Iam using it as Usage of ktlint says but it fails.. Why there is no source demo project with integrated ktlint ? It would be much better to immediately understand the setup and everything else...

```
$ ./gradlew ktlint --apply-to-idea-project -android
Parallel execution is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :ktlint from command line.
> Unknown command-line option '--apply-to-idea-project'.

```

**Comment by shyiko** [comment]
ktlint [returns exit code 1](https://github.com/shyiko/ktlint/issues/268#issuecomment-416681622) whenever there are style violations (in other words `ktlint ...` fails too (which is expected), you just don't see it). If you want build to succeed regardless of whether there are lint errors or not use [ignoreExitValue=true](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:ignoreExitValue).

Instead of `$ ./gradlew ktlint --apply-to-idea-project --android` you'll need something like:

```groovy
task ktlintApplyToIdeaProject(type: JavaExec, group: "formatting") {
    description = "Update Intellij IDEA project settings."
    classpath = configurations.ktlint
    main = "com.github.shyiko.ktlint.Main"
    args "--apply-to-idea-project", "--android"
}
```
(`./gradlew ktlintApplyToIdeaProject` to execute)

Another option is to use [one of the plugins](https://github.com/shyiko/ktlint#with-a-plugin).
